File: tests/examples/python_fmu/resources/fmi2.py
class Fmi2Status:
    """Represents the status of the FMU or the results of function calls.

    Values:
        * ok: all well
        * warning: an issue has arisen, but the computation can continue.
        * discard: an operation has resulted in invalid output, which must be discarded
        * error: an error has ocurred for this specific FMU instance.
        * fatal: an fatal error has ocurred which has corrupted ALL FMU instances.
        * pending: indicates that the FMu is doing work asynchronously, which can be retrived later.

    Notes:
        FMI section 2.1.3

    """

    ok = 0
    warning = 1
    discard = 2
    error = 3
    fatal = 4
    pending = 5


class FMU:
    def __init__(self) -> None:
        pass

    def set_debug_logging(self, categories, logging_on):
        return Fmi2Status.ok

    def do_step(self, current_time: float, step_size: float, no_step_prior: bool):
        return Fmi2Status.ok

    def setup_experiment(self, tolerance=None, start_time=None, stop_time=None):
        return Fmi2Status.ok

    def enter_initialization_mode(self):
        return Fmi2Status.ok

    def exit_initialization_mode(self):
        return Fmi2Status.ok

    def terminate(self):
        return Fmi2Status.ok

    def reset(self):
        return Fmi2Status.ok

File: wrapper/cli/assets/java/src/main/java/fmu/Fmi2.java
package fmu;

enum FMI2Status {
    OK, WARNING, DISCARD, ERROR, FATAL, PENDING,
}

class FMI2FMU {
    /*
     * Base class defining default implementations for FMI2 related methods
     */
    public FMI2Status doStep(double currentTime, double stepSize, boolean noStepPrior) {
        return FMI2Status.OK;
    }

    public FMI2Status setupExperiment(double startTime, Double stopTime, Double tolerance) {
        return FMI2Status.OK;
    }

    public FMI2Status setDebugLogging(String[] categories, boolean loggingOn) {
        return FMI2Status.OK;
    }

    public FMI2Status enterInitializationMode() {
        return FMI2Status.OK;
    }

    public FMI2Status exitInitializationMode() {
        return FMI2Status.OK;
    }

    public FMI2Status terminate() {
        return FMI2Status.OK;
    }

    public FMI2Status reset() {
        return FMI2Status.OK;
    }

    public FMI2Status cancelStep() {
        return FMI2Status.OK;
    }

    public byte[] serialize() throws RuntimeException {
        throw new RuntimeException(
                "the serialization method is not defined. Either define this method in the subclass, or set the 'canGetAndSetFMUstate' and 'canGetAndSetFMUstate' to false in the modelDescription.xml to indicate that the FMU does not support this functionality");
    }

    public FMI2Status deserialize(byte[] state) throws RuntimeException {
        throw new RuntimeException(
                "the deserialization method is not defined. Either define this method in the subclass, or set the 'canGetAndSetFMUstate' and 'canGetAndSetFMUstate' to false in the modelDescription.xml to indicate that the FMU does not support this functionality");
    }
}
